File: CodingSamples/Java/Runtime/ReflectionTest2/src/PersonalLoan.java
package finance;


public class PersonalLoan {

	@MaxDuration(6)
	public double common(double amount, int period) {
		return 11 + 0.5 * (period / 3);
	}

	public double employee(double amount, int period) {
		return common(amount, period) - 5;
	}
}

File: CodingSamples/DotNet/Runtime/ReflectionTest2/BankLib/PersonalLoan.cs
namespace Finance;

public class PersonalLoan
{
    [MaxDuration(Limit = 6)]
    public double Common(double amount, int period) => 11 + 0.5 * (period / 3);

    public double Employee(double amount, int period) => 0.75 * Common(amount, period); 

    public double Senior(double amount, int period) => Common(amount, period) - 2;
}
